### PR TITLE
Add destroyed column to BookmarkLocation table (##3729)

### LINKED ISSUE #3729: Update bookmarkLocationDao to include "destroyed" info.
**Summary:** 
Currently we don't have a destroyed column in the table of bookmarkLocationDao. As you know pic and destroyed are added later. This can cause an issue when user bookmarked a destroyed place. Since there is no destroyed info stored in bookmark dao (currently null), pin will appear as star inside blue, instead of star inside gray. Even if this is an edge case, we better to cover it.

So we need to update bookmark db from 11 to 12 I guess. 

Disclaimer: Probably not a beginner friendly task.

### PR DESCRIPTION
**Pull request by gvaibhav1734**
**Description (required)**
Fixes #3729 

**What changes did you make and why?**
1. Database version changed from **12** to **13**
2. Migration to add column `location_destroyed` to BookmarkLocation table
3. Updated Tests and Dao to reflect the new column 

**Tests performed (required)**
Tested {prodDebug, betaDebug} on {Pixel 2 emulator} with API level {28}.
Checked for successful database migration (on emulator) as well.

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/bookmarks/locations/BookmarkLocationsDao.java [hunk 1]
```diff
@@ -276,6 +280,15 @@ public static void onUpdate(SQLiteDatabase db, int from, int to) {
                 }
                 return;
             }
+            if (from == 12 && to == 13) {
```
Comment (macgills) [inline app/src/main/java/fr/free/nrw/commons/bookmarks/locations/BookmarkLocationsDao.java:283]: this `from == x && to == y` is not how it should be done, typically this is done in a switch on `from` and you just let it fallthrough.

The problem is that we can never guarantee a user upgrades monotonically, a user could theoretically go from v1 to v100 with none of the intervening updates and migrations should happily walk them along that path

**Comment by macgills** [comment]
@ashishkumar468 are there any further tickets for Room migration? It isn't very beneficial to the project to keep maintaining these unwieldy classes

**Comment by ashishkumar468** [comment]
@macgills Not that I know of, this should be it :)

File: app/src/main/java/fr/free/nrw/commons/bookmarks/locations/BookmarkLocationsDao.java (hunk 1 above)
Comment (gvaibhav1734) [reply app/src/main/java/fr/free/nrw/commons/bookmarks/locations/BookmarkLocationsDao.java:283]: So should I go ahead and change it to switch with fall through?

Comment (macgills) [reply app/src/main/java/fr/free/nrw/commons/bookmarks/locations/BookmarkLocationsDao.java:283]: Yes please

**Comment by macgills** [comment]
@ashishkumar468 I'll create a ticket for converting each remaining ContentProvider so?

**Comment by ashishkumar468** [comment]
Yes that would be great, thanks :)

**Pull request merged**